sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -116,13 +116,14 @@ namespace Azure { namespace Core { namespace IO {
    */
   class MemoryBodyStream final : public BodyStream {
   private:
-    const uint8_t* m_data;
     int64_t m_length;
     int64_t m_offset = 0;
 
     int64_t OnRead(uint8_t* buffer, int64_t count, Azure::Core::Context const& context) override;
 
   public:
+    const uint8_t* m_data;
+
```

Comment:
Yes, if we want to do this, I would try to do something like friend (though I don't like using `friend`), or maybe expose the buffer publicly via a method or some other way rather than making the field public.

Reply:
@ahsonkhan, not specifically to you, but I think we should change our stance on friends. They are C++ language tools, specifically for this purpose - similar to C#'s `internal`. In my opinion, it is ok to friend classes within the library.
We are going to have the need for it anyway, but we'll write more code to do the same.
If friend declarations allow us to achieve what we need, in simple way that reflects the reality, while public API surface looks good, we should be ok too.

Reply:
I am open to evolve my understanding on it, and I know there are some useful scenarios for it (such as for testing), so maybe we make such decisions thoughtfully on a case-by-case basis, but wanted to share some potential pitfalls around tight coupling, that only show up as issues after multiple releases:
https://softwareengineering.stackexchange.com/a/132453

Another fun discussion:
https://stackoverflow.com/questions/521754/when-to-use-friend-class-in-c